packages/trie/src/trie.ts

```diff
@@ -113,6 +113,50 @@ export class Trie {
     return new Trie(opts)
   }
 
+  static async fromProof(proof: Proof, opts?: TrieOpts): Promise<Trie> {
```

Comment:
What if I want to import two proofs? Or, I want to import a new proof in my trie later? This is now not possible anymore, right?

Reply:
Ok, let's bring this back to the drawing board and first decide if this is a change we want respectively which makes sense and is thought through to the end.

Reply:
Will give this a "blocked" label for now.

Reply:
(this change (i.e. make `fromProof` static) does not seem to be referred in the linked issue)

Reply:
> What if I want to import two proofs? Or, I want to import a new proof in my trie later? This is now not possible anymore, right?

It seems like these tasks are made possible with the new `updateFromProof` function.

Reply:
Ah yes you are right, I somehow missed that.

Reply:
Feel free to "unblock" this again or course of you guys think you've gotten answers for most of the questions here and things can be introduced in a non breaking way, you guys have a lot better overview than me here what's useful and makes sense regarding all this proof stuff!

Reply:
yes, `updateFromProof` is just the old `fromProof` with the word `update` tagged on to distinguish from the other.  To avoid actual breaking changes, we could instead use `createFromProof` for the static method, and leave `fromProof` as it was.

Reply:
We now have https://github.com/ethereumjs/ethereumjs-monorepo/pull/3186. I think we want to make the methods static in some cases? (But this is breaking)

Reply:
Yes, the goal is to have a static version of `verifyProof` so that a new trie instantiation isn't needed for each proof verification, or even more awkward is the proofTrie pattern where we keep a trie around in both the statemanager and snap sync fetchers to perform proof verification.

The changes fro #3186 should be included, so we should have a static and non-static version of `verifyProof` and `fromProof`. This shouldn't be breaking, right? Since users can just use the previously available non-static versions if they chose to.